azure-proto-core-test/IdentityTests.cs

```diff
@@ -0,0 +1,107 @@
+        public void CheckNullProperty()
+        {
+            Identity identity1 = new Identity();
+            identity1.ResourceId = "/subscriptions/6b085460-5f21-477e-ba44-1035046e9101/resourceGroups/nbhatia_test/providers/Microsoft.Web/sites/autoreport";            
+            Assert.IsNotNull(identity1.ResourceId);
+            Identity identity2 = new Identity();
+            Assert.IsNull(identity2.ResourceId);
+        }
+        
+        [TestCase("/subscriptions/6b085460-5f21-477e-ba44-1035046e9101/resourceGroups/nbhatia_test/providers/Microsoft.Web/sites/autoreport", "/subscriptions/6b085460-5f21-477e-ba44-1035046e9101/resourceGroups/nbhatia_test/providers/Microsoft.Web/sites/autoreport")]
+        [TestCase("/subscriptions/6b085460-5f21-477e-ba44-1035046e9101", "/subscriptions/6b085460-5f21-477e-ba44-1035046e9101")]
+        [TestCase("/subscriptions/6b085460-5f21-477e-ba44-1035046e9101/resourceGroups/nbhatia_test", "/subscriptions/6b085460-5f21-477e-ba44-1035046e9101/resourceGroups/nbhatia_test")]
+        public void CompareToZero(string resourceId1, string resourceId2)
+        {
+            Identity identity1 = new Identity();
+            identity1.ResourceId = resourceId1;
+            Identity identity2 = new Identity();
+            identity2.ResourceId = resourceId2;
+            Assert.AreEqual(0, identity1.CompareTo(identity2));
+        }
+
+        [TestCase("/subscriptions/6b085460-5f21-477e-ba44-1035046e9101/resourceGroups/nbhatia_test/providers/Microsoft.Web/sites/autoreport", "/subscriptions/6b085460-5f21-477e-ba44-1035046e9101", false)]
+        [TestCase("/subscriptions/6b085460-5f21-477e-ba44-1035046e9101", null, true)]
+        public void CompareToOne(string resourceId1, string resourceId2, bool isNull)
+        {
+            Identity identity1 = new Identity();
+            identity1.ResourceId = resourceId1;
+            Identity identity2 = new Identity();
+            identity2.ResourceId = resourceId2;
+            Assert.AreEqual(1, identity1.CompareTo(isNull?null: identity2));           
+        }
+
+        [TestCase("/subscriptions/6b085460-5f21-477e-ba44-1035046e9101", "/subscriptions/6b085460-5f21-477e-ba44-1035046e9101/resourceGroups/nbhatia_test")]
+        [TestCase("/subscriptions/6b085460-5f21-477e-ba44-1035046e9101", "/subscriptions/6b085460-5f21-477e-ba44-1035046e9101/resourceGroups/nbhatia_test/providers/Microsoft.Web/sites/autoreport")]
+        public void CompareToMinusOne(string resourceId1, string resourceId2)
+        {
+            Identity identity1 = new Identity();
+            identity1.ResourceId = resourceId1;
+            Identity identity2 = new Identity();
+            identity2.ResourceId = resourceId2;
+            Assert.AreEqual(-1, identity1.CompareTo(identity2));
+        }
+
+        [TestCase]
+        public void EqualsMethodTrue()
+        {
+            Identity identity1 = new Identity();
+            Identity identity2 = new Identity();
+            
```

Comment:
See comment below on equivalence classes, which would apply to equivalence and non-equivalence